qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/create/CreateOfferDataSource.groovy

```diff
@@ -0,0 +1,33 @@
+package life.qbic.portal.portlet.offers.create
+
+import life.qbic.datamodel.dtos.business.Offer
+import life.qbic.portal.portlet.exceptions.DatabaseQueryException
+
+/**
+ * Provides methods to store offers in a data-source
+ *
+ * @author Sven Fillinger
+ * @since 1.0.0
+ */
+interface CreateOfferDataSource {
+
+  /**
+   * Saves an offer in a persistent data-source.
+   *
+   * Developers should execute this method, when they
+   * want to store an offer in f.e. a database.
+   *
+   * If the passed offer is already in the database (offer
+   * content with the same offer identifier already exists),
+   * then this method will throw a DatabaseQueryException.
+   *
+   * If the passed offer cannot be saved successfully, this method
+   * will also throw a DatabaseQueryException.
+   *
+   * @param offer
+   * @return
+   * @throws DatabaseQueryException
+   */
+  save(Offer offer) throws DatabaseQueryException
```

Comment:
I would like to keep the SQL out of the domain level. So maybe throw a more generic Exception here or make DatabaseQueryException not extend SQL specific exceptions.
```suggestion
  save(Offer offer) throws IOException
```

Reply:
I fixed the DataBaseException instead, as the IOException is to generic imo. We now force the implementation to throw a DatabaseException in any case, which makes exception handling straight forward in the use case. IO Exceptions should not be part of the use case implementation, as it is already an implementation detail